topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -344,11 +347,73 @@ public TopicOperator(Vertx vertx, Kafka kafka,
         this.vertx = vertx;
         this.labels = labels;
         this.topicStore = topicStore;
-        this.inFlight = new InFlight<>(vertx);
         this.namespace = namespace;
         this.config = config;
     }
 
+    /**
+     * Run the given {@code action} on the context thread,
+     * immediately if there are currently no other actions with the given {@code key},
+     * or when the other actions with the given {@code key} have completed.
+     * When the given {@code action} is complete it must complete its argument future,
+     * which will complete the returned future
+     */
+    public <T> Future<T> acquireLock(TopicName key, Handler<Future<T>> action) {
```

Comment:
about the name, it doesn't just acquire a lock but even add the action to do in the queue. Maybe just an `enqueueAction`  ? From the code it's clear it acquire a lock for doing that maybe not need to reflect in the name?

Reply:
I agree `acquireLock` is a terrible name, but I couldn't think of a better one. How about `executeWithTopicLockHeld`, which does exactly what it says, even though it's a bit long. Wdyt?